golem-worker-service-base/src/evaluator/evaluator_context.rs

```diff
@@ -0,0 +1,69 @@
+use crate::evaluator::getter::GetError;
+use crate::evaluator::path::Path;
+use crate::evaluator::Getter;
+use crate::merge::Merge;
+use crate::worker_binding::RequestDetails;
+use crate::worker_bridge_execution::{RefinedWorkerResponse, WorkerRequest};
+use golem_wasm_rpc::TypeAnnotatedValue;
+
+// Evaluator of an expression doesn't necessarily need a context all the time, and can be empty.
+// or contain worker details, request details, worker_response or all of them.
+
+#[derive(Clone)]
+pub struct EvaluationContext {
```

Comment:
This is very poorly typed, in my opinion, because any combination of things can be `None`. Structures like this are usually better off as `enum`, but in this case, I will suggest something different for `evaluate` below.

Reply:
@jdegoes  Initially this was an enum, but it can be any combination at any point in time,including empty. We can call evaluator with just an input Expression `Expr` without passing any extra context (context::empty)